src/Id.ts

```diff
@@ -8,26 +9,68 @@ import * as utils from './utils';
 type Id = IdInternal & number;
 
 class IdInternal extends Uint8Array {
-  public static create(): Id;
-  public static create(length: number): Id;
-  public static create(array: ArrayLike<number> | ArrayBufferLike): Id;
-  public static create(
+  public static create<T extends Id = Id>(): T;
+  public static create<T extends Id = Id>(id: T): T;
+  public static create<T extends Id = Id>(length: number): T;
+  public static create<T extends Id = Id>(
+    array: ArrayLike<number> | ArrayBufferLike,
+  ): T;
+  public static create<T extends Id = Id>(
     buffer: ArrayBufferLike,
     byteOffset?: number,
     length?: number,
-  ): Id;
-  public static create(...args: Array<any>): Id {
+  ): T;
+  public static create<T extends Id = Id>(...args: Array<any>): T {
     // @ts-ignore: spreading into Uint8Array constructor
-    return new IdInternal(...args) as Id;
+    return new IdInternal(...args) as T;
   }
 
   public [Symbol.toPrimitive](_hint: 'string' | 'number' | 'default'): string {
     return utils.toString(this as unknown as Id);
   }
 
+  public static fromString<T extends Id = Id>(idString: string): T | undefined {
+    return utils.fromString(idString) as T;
+  }
+
+  /**
+   * Decodes as Buffer zero-copy
+   */
+  public static fromBuffer<T extends Id = Id>(idBuffer: Buffer): T | undefined {
+    return utils.fromBuffer(idBuffer) as T;
+  }
+
+  public static fromUUID<T extends Id = Id>(uuid: string): T | undefined {
```

Comment:
Are these generics still needed anymore? I imagine they won't be needed since we aren't using it anymore.

Reply:
They're for streamlining creating the Opaque types. see the comment below.